image-ext2.c

```diff
@@ -144,12 +150,10 @@ static int ext2_setup(struct image *image, cfg_t *cfg)
 
 	ext->features = cfg_getstr(image->imagesec, "features");
 	if (!ext->features) {
-		if (!ext->use_mke2fs) {
-			if (!strcmp(image->handler->type, "ext3"))
-				ext->features = "has_journal";
-			else if (!strcmp(image->handler->type, "ext4"))
-				ext->features = "extents,uninit_bg,dir_index,has_journal";
-		}
+		if (!strcmp(image->handler->type, "ext3"))
+			ext->features = "has_journal";
+		else if (!strcmp(image->handler->type, "ext4"))
+			ext->features = "extents,uninit_bg,dir_index,has_journal";
```

Comment:
No, this should be left as it is. These options are a crude imitation of what ext3/ext4 means because `genext2fs` only supports ext2. The options where chosen based on the defaults that `mke2fs` used at the time and never updated.

With `mke2fs` `mke2fs.conf` provides the correct defaults.

Reply:
> With `mke2fs` `mke2fs.conf` provides the correct defaults.

Okay so it probably uses `sysroot-host/etc/mke2fs.conf` then? Well, I'll send a v2 tomorrow.

Reply:
> Okay so it probably uses `sysroot-host/etc/mke2fs.conf` then? Well, I'll send a v2 tomorrow.

We're talking about PTXdist here, right? It was supposed to, but I lost that somewhere... :-/. I'm working on a fix. In the meantime (and for existing releases) `mke2fs-conf = "@PTXDIST_SYSROOT_HOST@/etc/mke2fs.conf"` in the genimage config should work.

Reply:
Yes, some layers on top of DistroKit with ptxdist 2019.11.0 atm. The file is almost identical to the one on my host machine (only difference is features for *ext4dev*), so I leave it as is for now. It's not scope of this PR anyway, or is it?

Reply:
No, it's something I need to fix in PTXdist anyways. I just mentioned it in case that causes additional problems.